app/src/main/java/com/kickstarter/viewmodels/MessagesViewModel.java

```diff
@@ -75,129 +103,211 @@ public ViewModel(final @NonNull Environment environment) {
       this.client = environment.apiClient();
       this.currentUser = environment.currentUser();
 
-      final Observable<MessageThread> messageThread = intent()
-        .take(1)
-        .map(i -> i.getParcelableExtra(IntentKey.MESSAGE_THREAD))
-        .ofType(MessageThread.class);
+      final Observable<Either<MessageThread, Pair<Project, Backing>>> configData = intent()
+        .map(i -> {
+          final MessageThread messageThread = i.getParcelableExtra(IntentKey.MESSAGE_THREAD);
+          return messageThread != null
+            ? new Either.Left<MessageThread, Pair<Project, Backing>>(messageThread)
+            : new Either.Right<MessageThread, Pair<Project, Backing>>(
+              Pair.create(i.getParcelableExtra(IntentKey.PROJECT), i.getParcelableExtra(IntentKey.BACKING))
+          );
+        });
+
+      final Observable<KoalaContext.Message> koalaContext = intent()
+        .map(i -> i.getSerializableExtra(IntentKey.KOALA_CONTEXT))
+        .ofType(KoalaContext.Message.class);
+
+      final Observable<Backing> configBacking = configData
+        .map(Either::right)
+        .filter(ObjectUtils::isNotNull)
+        .map(PairUtils::second);
+
+      final Observable<MessageThread> configThread = configData
+        .map(Either::left)
+        .filter(ObjectUtils::isNotNull);
+
+      final Observable<Either<Backing, MessageThread>> backingOrThread = Observable.merge(
+        configBacking.map(backing -> new Either.Left<>(backing)),
```

Comment:
this can be replaced with a method reference, as can the `thread -> new Either.Right<>(thread)` below, but I find this syntax more understandable than

```java
(Func1<Backing, Either.Left<Backing, MessageThread>>) Either.Left::new
```
if that's okay